src/librustc/ty/context.rs

```diff
@@ -1604,20 +1611,23 @@ impl<'a, 'tcx> TyCtxt<'a, 'tcx, 'tcx> {
     }
 }
 
-impl<'gcx: 'tcx, 'tcx> GlobalCtxt<'gcx> {
+impl<'gcx> GlobalCtxt<'gcx> {
     /// Call the closure with a local `TyCtxt` using the given arena.
```

Comment:
Can we add a comment here:

---

The `interners` parameter is a reference to a storage slot with `'tcx` lifetime that we can use. It is expected to be `None` on entry and will be reset to `None` on return.

---

that presumes, of course, that you accept my proposal below =)